blockchain/types/gen_header_json.go

```diff
@@ -16,22 +16,24 @@ var _ = (*headerMarshaling)(nil)
 // MarshalJSON marshals as JSON.
 func (h Header) MarshalJSON() ([]byte, error) {
 	type Header struct {
-		ParentHash  common.Hash    `json:"parentHash"       gencodec:"required"`
-		Rewardbase  common.Address `json:"reward"           gencodec:"required"`
-		Root        common.Hash    `json:"stateRoot"        gencodec:"required"`
-		TxHash      common.Hash    `json:"transactionsRoot" gencodec:"required"`
-		ReceiptHash common.Hash    `json:"receiptsRoot"     gencodec:"required"`
-		Bloom       Bloom          `json:"logsBloom"        gencodec:"required"`
-		BlockScore  *hexutil.Big   `json:"blockScore"       gencodec:"required"`
-		Number      *hexutil.Big   `json:"number"           gencodec:"required"`
-		GasUsed     hexutil.Uint64 `json:"gasUsed"          gencodec:"required"`
-		Time        *hexutil.Big   `json:"timestamp"        gencodec:"required"`
-		TimeFoS     hexutil.Uint   `json:"timestampFoS"              gencodec:"required"`
-		Extra       hexutil.Bytes  `json:"extraData"                 gencodec:"required"`
-		Governance  hexutil.Bytes  `json:"governanceData"            gencodec:"required"`
-		Vote        hexutil.Bytes  `json:"voteData,omitempty"`
-		BaseFee     *hexutil.Big   `json:"baseFeePerGas,omitempty"    rlp:"optional"`
-		Hash        common.Hash    `json:"hash"`
+		ParentHash   common.Hash    `json:"parentHash"       gencodec:"required"`
+		Rewardbase   common.Address `json:"reward"           gencodec:"required"`
+		Root         common.Hash    `json:"stateRoot"        gencodec:"required"`
+		TxHash       common.Hash    `json:"transactionsRoot" gencodec:"required"`
+		ReceiptHash  common.Hash    `json:"receiptsRoot"     gencodec:"required"`
+		Bloom        Bloom          `json:"logsBloom"        gencodec:"required"`
+		BlockScore   *hexutil.Big   `json:"blockScore"       gencodec:"required"`
+		Number       *hexutil.Big   `json:"number"           gencodec:"required"`
+		GasUsed      hexutil.Uint64 `json:"gasUsed"          gencodec:"required"`
+		Time         *hexutil.Big   `json:"timestamp"        gencodec:"required"`
+		TimeFoS      hexutil.Uint   `json:"timestampFoS"              gencodec:"required"`
+		Extra        hexutil.Bytes  `json:"extraData"                 gencodec:"required"`
+		Governance   hexutil.Bytes  `json:"governanceData"            gencodec:"required"`
+		Vote         hexutil.Bytes  `json:"voteData,omitempty"`
+		BaseFee      *hexutil.Big   `json:"baseFeePerGas,omitempty" rlp:"optional"`
+		RandomReveal []byte         `json:"randomReveal,omitempty" rlp:"optional"`
```

Comment:
Why not follow the convention `hexutil.Bytes`?

Reply:
Regenerated gen_header_json.